core/providers/elixir/elixir.go

```diff
@@ -0,0 +1,233 @@
+		miseStep.Version(elixir, strings.TrimSpace(string(versionFile)), ".elixir-version")
+	}
+
+	if mixExs, err := ctx.App.ReadFile("mix.exs"); err == nil {
+		if match := elixirVersionRegex.FindStringSubmatch(mixExs); len(match) > 2 {
+			version := utils.ExtractSemverVersion(match[2])
+			if version != "" {
+				fmt.Println("mix.exs", version)
+				miseStep.Version(elixir, version, "mix.exs")
+			}
+		}
+	}
+
+	pkgs, err := miseStep.Resolver.ResolvePackages()
+	erlang := miseStep.Default("erlang", DEFAULT_ERLANG_VERSION)
+	elixirVersion := DEFAULT_ELIXIR_VERSION
+	if err == nil && pkgs["elixir"] != nil && pkgs["elixir"].ResolvedVersion != nil {
+		elixirVersion = *pkgs["elixir"].ResolvedVersion
+	}
+
+	elixirSemverVersion := utils.ExtractSemverVersion(elixirVersion)
+	elixirSemver, err := utils.ParseSemver(elixirSemverVersion)
+
+	if err == nil {
+		compatibleErlangVersion := getCompatibleErlangVersion(fmt.Sprintf("%d.%d", elixirSemver.Major, elixirSemver.Minor))
+		miseStep.Version(erlang, compatibleErlangVersion, "default compatible OTP version")
+	}
+
+	if envVersion, varName := ctx.Env.GetConfigVariable("ERLANG_VERSION"); envVersion != "" {
+		miseStep.Version(erlang, envVersion, varName)
+	}
+
+	if versionFile, err := ctx.App.ReadFile(".erlang-version"); err == nil {
+		miseStep.Version(erlang, strings.TrimSpace(string(versionFile)), ".erlang-version")
+	}
+
+	versionParts := strings.Split(elixirVersion, "-otp-")
+	if len(versionParts) > 1 {
+		otpVersion := versionParts[1]
+		otpSemverVersion := utils.ExtractSemverVersion(otpVersion)
+		if _, err := utils.ParseSemver(otpSemverVersion); err == nil {
+			miseStep.Version(erlang, otpSemverVersion, "resolved compatible OTP version")
+		}
+	}
+}
+
+func (p *ElixirProvider) GetEnvVars(ctx *generate.GenerateContext) map[string]string {
+	return map[string]string{
+		"LANG":               "en_US.UTF-8",
+		"LANGUAGE":           "en_US:en",
+		"LC_ALL":             "en_US.UTF-8",
+		"ELIXIR_ERL_OPTIONS": "+fnu",
+		"MIX_ENV":            "prod",
+	}
+}
+
+func (p *ElixirProvider) findBinName(ctx *generate.GenerateContext) string {
+	configFile, err := ctx.App.ReadFile("mix.exs")
+	if err != nil {
+		return ""
+	}
+
+	scanner := bufio.NewScanner(strings.NewReader(configFile))
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.Contains(line, "app: :") {
+			binName := strings.Split(strings.Replace(line, "app:", "", 1), ":")[1]
+			binName = strings.TrimSpace(strings.Trim(binName, ",'\""))
+			return binName
+		}
+	}
+
+	if err := scanner.Err(); err != nil {
+		return ""
+	}
+
+	return ""
+}
+
+// See: https://hexdocs.pm/elixir/1.18.3/compatibility-and-deprecations.html#between-elixir-and-erlang-otp
+func getCompatibleErlangVersion(elixirVersion string) string {
+	switch elixirVersion {
+	case "1.0", "1.1":
+		return "18"
+	case "1.2", "1.3":
+		return "19"
+	case "1.4":
+		return "20"
+	case "1.5":
+		return "20"
+	case "1.6":
+		return "2`"
```

Comment:
This looks like a typo? I assume it should be `21`

Reply:
Yep! Thank you for reviewing